### PR TITLE
Durations populated from calibrations

### PR DESCRIPTION
**Pull request by lcapelluto**
### Summary
Circuits with custom gate calibrations can now be scheduled with the
transpiler without explicitly providing the durations of each circuit
calibration.

### Details and comments

I ran into this issue trying to use Naoki's timeline drawer with circuits that had calibrations. Realized I had missed updating this when pulse gates were introduced

### CODE REVIEW

File: qiskit/compiler/transpile.py [hunk 1]
```diff
@@ -419,7 +419,7 @@ def _parse_transpile_args(circuits, backend,
     routing_method = _parse_routing_method(routing_method, num_circuits)
     translation_method = _parse_translation_method(translation_method, num_circuits)
     durations = _parse_instruction_durations(backend, instruction_durations, dt,
-                                             num_circuits)
+                                             circuits, num_circuits)
```
Comment (itoko) [inline qiskit/compiler/transpile.py:422]: I think we can replace `num_circuits` with `circuits`.

File: qiskit/compiler/transpile.py [hunk 2]
```diff
@@ -623,20 +623,28 @@ def _parse_scheduling_method(scheduling_method, num_circuits):
...
-                                         dt or getattr(inst_durations, 'dt', None))
+def _parse_instruction_durations(backend, inst_durations, dt, circuits, num_circuits):
+    backend_durations = InstructionDurations()
+    try:
+        backend_durations = InstructionDurations.from_backend(backend)
+    except AttributeError:
+        pass
+
+    durations = []
+    for i in range(num_circuits):
```
Comment (itoko) [inline qiskit/compiler/transpile.py:634]: We can now iterate over `circuits` instead of `num_circuits`.

**Review by itoko — CHANGES_REQUESTED** [review]
Thank you for taking on this fix.

It seems to revert the spec change introduced in #5164 unintentionally.
According to #5164, `backend_durations` should be "overwritten" by `inst_durations` rather than "updated" as done in this PR.
I think, for `cal_durations`, updating would be more suitable than overwriting since cal gates are not args in transpile(), though.

**Comment by lcapelluto** [comment]
Great points! I've updated

**Review by itoko — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**